sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -218,6 +219,17 @@ public TimeSpan MaxWaitTime
         /// </summary>
         public InitialOffsetOptions InitialOffsetOptions { get; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether the trigger will create
+        /// checkpoints as events are being processed.  The default value is <c>true</c>.
+        /// </summary>
+        /// <value><c>true</c> if checkpoints will be written; otherwise, <c>false</c>.</value>
+        /// <remarks>
+        /// This value takes precedence over <see cref="BatchCheckpointFrequency"/>, which will be
+        /// ignored if this property is set to <c>false</c>.
+        /// </remarks>
+        public bool EnableCheckpointing { get; set; } = true;
```

Comment:
At one point I remember we talked about potentially allowing this on a per-invocation basis - do we not need that anymore based on the recent changes to no longer checkpoint when shutting down?

Reply:
It may be something that we eventually want to add to the context - but I don't have a strong use case for it at the moment.    I think that if we do end up adding a per-invocation override, it would fit nicely on the context and be complimentary to this.    

This one is a valid but unusual case where events are transient shoulder-tap style notifications and anything more than a couple of minutes old is no longer interesting, so checkpoints were just slowing things down and getting in the way.